docs: fix step name in long-running workflow documentation

Vallabh-Mahanta opened the pull request:
Fixed inconsistency in step name - updated 'set-step-failure-step' to 'setStepFailureStep' to match the exported function name.

Fixes #12143

Commit:
docs: fix step name in long-running workflow documentation

Fixed inconsistency in the step name - updated 'set-step-failure-step' to 'setStepFailureStep' to match the exported function name.

Fixes #12143

shahednasser commented on www/apps/book/app/learn/fundamentals/workflows/long-running-workflow/page.mdx:
```diff
@@ -252,7 +252,7 @@ type SetStepFailureStepInput = {
 
 export const setStepFailureStep = createStep(
-  "set-step-success-step",
+  "setStepFailureStep",
```
```suggestion
  "set-step-failure-step",
```

shahednasser reviewed (changes requested):
Thanks for contributing! Just a small change required

Vallabh-Mahanta committed:
docs: fix step name in long-running workflow documentation

Changed the step ID from "setStepFailureStep" to "set-step-failure-step" as requested to follow kebab-case naming convention for step IDs.

Fixes #12143

shahednasser reviewed (changes requested):
It looks like you accidentally removed all content, can you check your changes?

Vallabh-Mahanta committed:
Update page.mdx

I think this is done. Sorry for inconvenience caused by me.

Vallabh-Mahanta committed:
Update page.mdx

Sorry for the inconvenience.

Vallabh-Mahanta committed:
Update page.mdx

Sorry for the inconvenience.

Vallabh-Mahanta committed:
Update page.mdx

shahednasser reviewed (approved):
LGTM thanks!
